src/amp-story-player/amp-story-player-impl.js

```diff
@@ -774,21 +819,107 @@ export class AmpStoryPlayer {
   }
 
   /**
-   * Updates the muted state of the story inside the iframe.
+   * Updates the specified iframe's story state with given value.
+   * @param {number} iframeIdx
+   * @param {string} state
+   * @param {boolean} value
+   * @private
+   */
+  updateStoryState_(iframeIdx, state, value) {
+    this.messagingPromises_[iframeIdx].then((messaging) => {
+      messaging.sendRequest('setDocumentState', {state, value});
+    });
+  }
+
+  /**
+   * Update the muted state of the story inside the iframe.
    * @param {number} iframeIdx
    * @param {boolean} mutedValue
    * @private
    */
   updateMutedState_(iframeIdx, mutedValue) {
+    this.updateStoryState_(
+      iframeIdx,
+      STORY_MESSAGE_STATE_TYPE.MUTED_STATE,
+      mutedValue
+    );
+  }
+
+  /**
+   * Send message to story asking for page attachment state.
+   * @private
+   */
+  getPageAttachmentState_() {
+    const iframeIdx = this.stories_[this.currentIdx_][IFRAME_IDX];
     this.messagingPromises_[iframeIdx].then((messaging) => {
-      messaging.sendRequest(
-        'setDocumentState',
-        {state: 'MUTED_STATE', value: mutedValue},
-        true
-      );
+      messaging
+        .sendRequest(
+          'getDocumentState',
+          {state: STORY_MESSAGE_STATE_TYPE.PAGE_ATTACHMENT_STATE},
+          true
+        )
+        .then((event) => this.dispatchPageAttachmentEvent_(event.value));
     });
   }
 
+  /**
+   * React to documentStateUpdate events.
+   * @param {!DocumentStateTypeDef} data
+   * @private
+   */
+  onDocumentStateUpdate_(data) {
+    switch (data.state) {
+      case STORY_MESSAGE_STATE_TYPE.PAGE_ATTACHMENT_STATE:
+        this.onPageAttachmentStateUpdate_(data.value);
+        break;
+      default:
+        break;
+    }
+  }
+
+  /**
+   * React to page attachment update events.
+   * @param {boolean} pageAttachmentOpen
+   * @private
+   */
+  onPageAttachmentStateUpdate_(pageAttachmentOpen) {
+    this.updateButtonVisibility_(!pageAttachmentOpen);
+    this.dispatchPageAttachmentEvent_(pageAttachmentOpen);
+  }
+
+  /**
+   * Updates the visbility state of the exit control button.
+   * @param {boolean} isVisible
+   * @private
+   */
+  updateButtonVisibility_(isVisible) {
+    const button = this.rootEl_.querySelector(
+      'button.amp-story-player-exit-control-button'
+    );
+    if (button === null) {
```

Comment:
I think this should be `if (!button)`, right?  It should also short circuit if `querySelector` were to return `undefined`?

Reply:
Fixed!